File: update-core/src/test/java/dev/bee/kanjianki/updatecore/UpdateNotificationPolicyTest.java
package dev.bee.kanjianki.updatecore;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public final class UpdateNotificationPolicyTest {
    @Test
    public void pendingUpdateNotificationRequiresPermissionAndEnabledNotifications() {
        assertFalse(UpdateNotificationPolicy.shouldShowPendingUpdate(false, true));
        assertFalse(UpdateNotificationPolicy.shouldShowPendingUpdate(true, false));
        assertTrue(UpdateNotificationPolicy.shouldShowPendingUpdate(true, true));
    }
}

File: update-core/src/test/kotlin/dev/bee/kanjianki/updatecore/UpdateNotificationPolicyTest.kt
package dev.bee.kanjianki.updatecore

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class UpdateNotificationPolicyTest {
    @Test
    fun pendingUpdateNotificationRequiresPermissionAndEnabledNotifications() {
        assertFalse(UpdateNotificationPolicy.shouldShowPendingUpdate(false, true))
        assertFalse(UpdateNotificationPolicy.shouldShowPendingUpdate(true, false))
        assertTrue(UpdateNotificationPolicy.shouldShowPendingUpdate(true, true))
    }
}
